middle_end/flambda2/from_lambda/closure_conversion.ml

```diff
@@ -675,49 +678,51 @@ let close_named acc env ~let_bound_var (named : IR.named)
       else symbol_for_ident acc env id
     in
     let named = Named.create_simple simple in
-    k acc (Some named)
+    k acc env (Some named)
   | Simple (Const cst) ->
     let acc, named, _name = close_const acc cst in
-    k acc (Some named)
+    k acc env (Some named)
   | Get_tag var ->
     let named = find_simple_from_id env var in
     let prim : Lambda_to_flambda_primitives_helpers.expr_primitive =
       Unary (Tag_immediate, Prim (Unary (Get_tag, Simple named)))
     in
-    Lambda_to_flambda_primitives_helpers.bind_rec acc None
+    Lambda_to_flambda_primitives_helpers.bind_rec acc env None
       ~register_const_string:(fun acc -> register_const_string acc)
       prim Debuginfo.none
-      (fun acc named -> k acc (Some named))
+      (fun acc env named -> k acc env (Some named))
   | Begin_region ->
     let prim : Lambda_to_flambda_primitives_helpers.expr_primitive =
       Nullary Begin_region
     in
-    Lambda_to_flambda_primitives_helpers.bind_rec acc None
+    Lambda_to_flambda_primitives_helpers.bind_rec acc env None
       ~register_const_string:(fun acc -> register_const_string acc)
       prim Debuginfo.none
-      (fun acc named -> k acc (Some named))
+      (fun acc env named -> k acc env (Some named))
   | End_region id ->
     let named = find_simple_from_id env id in
     let prim : Lambda_to_flambda_primitives_helpers.expr_primitive =
       Unary (End_region, Simple named)
     in
-    Lambda_to_flambda_primitives_helpers.bind_rec acc None
+    Lambda_to_flambda_primitives_helpers.bind_rec acc env None
       ~register_const_string:(fun acc -> register_const_string acc)
       prim Debuginfo.none
-      (fun acc named -> k acc (Some named))
+      (fun acc env named -> k acc env (Some named))
   | Prim { prim; args; loc; exn_continuation } ->
     close_primitive acc env ~let_bound_var named prim ~args loc exn_continuation
       k
 
 let close_let acc env id user_visible defining_expr
     ~(body : Acc.t -> Env.t -> Expr_with_acc.t) : Expr_with_acc.t =
-  let body_env, var = Env.add_var_like env id user_visible in
-  let cont acc (defining_expr : Named.t option) =
+  (* CR keryan : We can avoid having the bound variable in the environment of
+     the defining expression but it should not appear there anyway *)
```

Comment:
Yes, we need it at least to register the pairing of the final `Box` of a primitive since the variable is not generated by `Lambda_to_flambda_primitive` as other intermediary variables.